backend_kvm.go

```diff
@@ -0,0 +1,136 @@
+// +build linux
+// +build amd64
+
+package fakemachine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type KvmBackend struct {}
+
+func init() {
+	RegisterBackend(KvmBackend{})
+}
+
+func (b KvmBackend) Name() string {
+	return "kvm"
+}
+
+func (b KvmBackend) Supported() (bool, error) {
+	if _, err := os.Stat("/dev/kvm"); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (b KvmBackend) InitScriptModules() []string {
+	return []string{"virtio_pci", "virtio_console", "9pnet_virtio", "9p"}
+}
+
+func (b KvmBackend) GetExtraVolumes() []mountPoint {
+	return nil
+}
+
+func (b KvmBackend) MountOptions(mount mountPoint) (fstype string, options []string, device string, mountpoint string) {
+	fstype = "9p"
+	options = []string{"trans=virtio", "version=9p2000.L", "cache=loose", "msize=262144"}
+	device = mount.label
+	mountpoint = mount.machineDirectory
```

Comment:
Any reason why this function need to define the mountpoint and the device/label ? I'd expect that's knowledge the machine has rather then the backend

Reply:
in the KVM 9p case, the volume is mounted by its label, but in the UML hostfs case the volume is mounted by a mount option and the device is none.
So the backend really needs to decide the fstype, device and options. But you are right; we could drop the mountpoint.